### ISSUE TITLE
SORT DESC clause not working in subqueries

### ISSUE THREAD

**Issue by ctmakro** [arangodb/arangodb#1942]
consider the following code.
collection `posts` have skiplist on [tid, toc]
```aql
    //specify a thread
    let t = document(threads,'80267')

    let lm = (
      FOR p IN posts
      FILTER p.tid == t._key //all post of that thread
      sort p.toc desc
      limit 1
      return p
    )
return lm
```
where `p.toc` means PostTimeOfCreation. Basically I want to sort by creation time and return the latest.

Surprisingly, above query returned the earliest post (the DESC didn't work). Our production site were screwed up (not really expecting this kind of bug...), now we are going to switch back to 2.8.

however, by moving FOR out of subquery:
```aql
      FOR p IN posts
      FILTER p.tid == '80267' //all post of that thread
      sort p.toc desc
      limit 2
      return p
```
above query worked fine.

**Comment by Simran-B** [comment]
Can you give some example values of toc?

**Comment by ctmakro** [comment]
@Simran-B toc is timestamp, or Date. 
Typical value would be something like `Date.now()` or `date_now()`

**Comment by ctmakro** [comment]
moar detail: server 2012, switched from 2.8.x to 3.0.2 about 24h ago. now ready to switch back.

**Comment by jsteemann** [comment]
Fixed in branches 3.0 and devel. The fix will become available in release 3.0.3.

**Comment by Simran-B** [comment]
@ctmakro: Apologies that we didn't discover this bug ourselves and sorry for all the inconvenience caused. There were tremendous changes made between 2.8 to 3.0 to the ArangoDB core, including a complete change of the internal data format, various performance optimizations and so on. This problem must have slipped in during this transition.

So thank you for discovering this, we and all probably all of our users appreciate it! The fix is already checked in, if you need it right now consider to compile ArangoDB yourself, or wait for the soon to be released bug-fix release.

In general, it is recommendable to do major version upgrades of your system in a safe test environment (copy of live system), then run own tests (should be automatic as well as some manual checks). If you're happy with the result and no problems occurred, make sure you have a backup of your live system before you upgrade it. It might be desired to keep the old system running and make a new, upgraded system ready for production use on other machines, then flip a switch to transition over to the new system. You should still announce a possible downtime to your users, and probably make the service temporarily unavailable from outside.

**Comment by jsteemann** [comment]
Fixed in 3.0.3
